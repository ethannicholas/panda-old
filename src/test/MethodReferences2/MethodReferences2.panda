@abstract
class Test {
    @class
    function a():String {
        return "I'm 'a'!"
    }

    method b():(String, String) {
        return ("I'm", "'b'!")
    }

    function c(a:()=>(String)):String {
        return a() + "... as interpreted by 'c1'!"
    }

    @class
    method c(s:String):String {
        return "And another variant of 'c' says '\{s}'!"
    }

    @abstract
    method d(s:String):String

    @external
    method e(s:String):String
}

interface TestInterface {
    method f()    
}

def c := Test.c
def d := Test.d
def e := Test.e
def f := TestInterface.f