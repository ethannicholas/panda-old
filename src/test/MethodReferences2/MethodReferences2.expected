error: MethodReferences2.panda:32:10: 'Test' has more than one method named 'c', specify parameter types to disambiguate
error: MethodReferences2.panda:33:10: 'Test.d' is abstract
error: MethodReferences2.panda:35:10: 'TestInterface.f' is abstract
3 errors
